### PR TITLE
Improve Search functionality

### PR DESCRIPTION
**Pull request by cstns**
## Description

- Replaced algolia's netlify search plugin with algolia's autocomplete.js lib.
- namespaced search bars with current page scope (currently only blog, handbook and docs scope used)
- Updated article nav entries for better indexing
- styling

## Related Issue(s)

Has to be accepted after https://github.com/FlowFuse/website/pull/2659 to avoid a broken blog search
closes https://github.com/FlowFuse/website/issues/2630

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

**Comment by cstns** [comment]
There will be a short period where the blog search bar will not work (will not return any results) until Algolia indexes the 'blog' keyword correctly.

Should I break up the addition of the blog keyword into a separate PR, release it before this one in order to avoid this?

**Comment by joepavitt** [comment]
> Should I break up the addition of the blog keyword into a separate PR, release it before this one in order to avoid this?

Very sensible

**Comment by cstns** [comment]
I created https://github.com/FlowFuse/website/pull/2659 that should update the indices. I'm not going to cherry pick the changes out of this PR, they should sort themselves out when https://github.com/FlowFuse/website/pull/2659 is merged in main and main in merge in the current PR.

After deploying https://github.com/FlowFuse/website/pull/2659, I'll check manually for the algolia indices to be updated and the deployment branch. Until then, I'll leave this PR marked as draft

File: src/support.njk [hunk 1]
```diff
@@ -81,20 +81,4 @@ hubspot:
...
-    algoliasearchNetlify({
-        appId: 'KWE0727TZS',
-        apiKey: '595ee07f069991380e9c3036cfad8e5e',
-        siteId: '00f8cf60-997f-4c4d-9427-a97924358648',
-        branch: 'live',
-        selector: 'div#algolia-search',
-        detached: false
-    });
-    }
-</script>
```
Comment (Yndira-E) [inline src/support.njk:100]: @cstns, I think we should include the `common-js.njk` file here, right?

It currently looks like this:
Screenshot 2024-10-09 at 14 35 02

Or did you intend to remove it entirely? It made sense when it was a global search.

Comment (cstns) [reply src/support.njk:100]: I'm looking into it, I can't tell right now why it's not working on the deployed branch and working just fine locally.

Also, should this search bar be scoped?

Comment (cstns) [reply src/support.njk:100]: nevermind, I was toggling between branches.

**Comment by cstns** [comment]
> After deploying #2659, I'll check manually for the algolia indices to be updated and the deployment branch. Until then, I'll leave this PR marked as draft

Algolia successfully updated the new indices
image

**Comment by joepavitt** [comment]
Is this ready for review @cstns?

**Comment by cstns** [comment]
The blog namespaced search should work now, but can't tell why the tests ([untitaker step](https://github.com/FlowFuse/website/actions/runs/11255980074/job/31296770812?pr=2639) ) are failing

**Comment by cstns** [comment]
After tackling the problem like a deaf man hitting a church bell, I was able to finally figure out why the untitaker/hyperlink checks were failing.

The problem laid in the plugin's inability to resolve dynamic urls, thus resolving/treating `<a href="${item.url}" ...` and `<img src="${item.image}" ...` to literal strings which of course are not url spec compliant.

Although there is a [closed issue](https://github.com/untitaker/hyperlink/issues/148) on the matter, for whatever reason script src's are not ignored in our case, even though we are using a version greater than the one containing the fix.

The only feasible workaround I could come up with is to add the url/src values as data attributes on the elements and swap them at runtime when the autocomplete query changes state.

**Comment by cstns** [comment]
@Yndira-E, I know you requested to have the search result title/heading bold but the autocomplete.js lib highlight does exactly that when highlighting results for content titles.

I can revisit the highlighting functionality when tackling the global search task if needed.

**Review by Yndira-E — APPROVED** [review]
Approving now and we'll review the styling for future iterations if needed.

**Pull request merged**